### PR TITLE
fix unescaped slash error in preg match

### PR DESCRIPTION
**Pull request by SaidkhojaIftikhor**
This is:

- [x] a bugfix
- [ ] a new feature
- [ ] refactoring
- [ ] additional unit tests

Checklist:

- [x] Changes are covered by unit tests
  - [x] Changes are covered by existing unit tests
  - [ ] New unit tests have been added
- [x] Code style is respected
- [x] Commit message explains **why** the change is made (see https://github.com/erlang/otp/wiki/Writing-good-commit-messages)
- [x] CHANGELOG.md contains a short summary of the change and a link to the pull request if applicable
- [ ] Documentation is updated as necessary

### Why this change is needed?

I faced with an error _<warning> WARNING </warning> preg_match(): Unknown modifier '�'_. 
After some debuging I found out that preg_quote doesn't escape slash character. I added slash character as a second parameter of preg_quote and now it works :)

image

[preg_quote](https://www.php.net/manual/en/function.preg-quote.php)

example code: This code will throw an error because `$columnName` has a slash in it and preg_quote will not escape it

```php
$columnName = 'Buy/Sell';
$columnId = 'C';
$reference = '[B5]';

$cellReference = $columnId . 5;
$pattern1 = '/\[' . preg_quote($columnName) . '\]/miu';
$pattern2 = '/@' . preg_quote($columnName) . '/miu';

if (preg_match($pattern1, $reference) === 1) {
    $reference = preg_replace($pattern1, $cellReference, $reference);
} elseif (preg_match($pattern2, $reference) === 1) {
    $reference = preg_replace($pattern2, $cellReference, $reference);
}
```

### CODE REVIEW

File: CHANGELOG.md
Comment (MarkBaker) [inline CHANGELOG.md:1]: I'm not sure why the entire ChangeLog has had an extra space added after every bullet point; throughout the entire history of the project

File: src/PhpSpreadsheet/Calculation/Engine/FormattedNumber.php
Comment (MarkBaker) [inline src/PhpSpreadsheet/Calculation/Engine/FormattedNumber.php:1]: I'm not sure that it's really necessary to escape `/` here; nowhere in the world uses a `/` as a decimal or thousands separator; and any developer that manually sets those locale separators to a `/` probably deserves everything that they get

File: src/PhpSpreadsheet/Calculation/Engine/Operands/StructuredReference.php
Comment (MarkBaker) [inline src/PhpSpreadsheet/Calculation/Engine/Operands/StructuredReference.php:1]: The crux of the PR; and I've verified that the change is necessary her.
Can you provide a unit test that will demonstrate the issue, and prove that this fix works?

File: src/PhpSpreadsheet/Cell/AdvancedValueBinder.php
Comment (MarkBaker) [inline src/PhpSpreadsheet/Cell/AdvancedValueBinder.php:1]: Can a `/` in a format mask drop through to this point, given the check for `/` in the fractional regexp immediately before?

File: src/PhpSpreadsheet/Calculation/TextData/Extract.php
Comment (MarkBaker) [inline src/PhpSpreadsheet/Calculation/TextData/Extract.php:1]: As with the Structured Reference changes, can you provide unit tests to verify that this is a problem, and that the solution does resolve that problem for all these instances where you've made the change?

File: CHANGELOG.md
Comment (SaidkhojaIftikhor) [reply CHANGELOG.md:1]: I've just merged master with my branch, this changes are from master

File: src/PhpSpreadsheet/Calculation/Engine/FormattedNumber.php
Comment (SaidkhojaIftikhor) [reply src/PhpSpreadsheet/Calculation/Engine/FormattedNumber.php:1]: Understand, I will remove them

File: src/PhpSpreadsheet/Calculation/Engine/Operands/StructuredReference.php
Comment (SaidkhojaIftikhor) [reply src/PhpSpreadsheet/Calculation/Engine/Operands/StructuredReference.php:1]: Ok, I will try. I am new to this repository and I don't understand a lot of things, that's why I added delimiter everywhere in code. 

How can I learn the codebase ?

**Comment by oleibman** [comment]
I looked into the testing that would be needed, and it seems pretty complicated, so complicated that I have just submitted 3 different PRs to handle it. If those PRs are merged, I will close this one.

**Comment by oleibman** [comment]
The replacement PRs are merged. Closing this one.

**Pull request closed without merging**